### PR TITLE
ui-slider

### PR DESCRIPTION
**Pull request by dalvarez**
Moving the ui-slider plugin into our addon dir and removing ember-cli-range-input.

N.B. This has the jquery rangeinput.js as a vendor file we have to app.import()

### CODE REVIEW

File: addon/templates/components/ui-slider.hbs [hunk 1]
```diff
@@ -1 +1,2 @@
-{{yield}}
+
+<input class="range-original" type="range" {{bind-attr min=min max=max step=step value=value}}>
```
Comment (twokul) [inline addon/templates/components/ui-slider.hbs]: no `bind-attr`s please

File: addon/components/ui-slider.js [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+
 export default Ember.Component.extend({
-  layout: layout
+  classNames: ['range-input'],
+  layout: layout,
+  min: 0,
+  max: 10,
+  step: 1,
+
+  value: computed(function (key, value) {
```
Comment (twokul) [inline addon/components/ui-slider.js]: missing `min` dependency key?

File: tests/dummy/app/templates/application.hbs [hunk 3]
```diff
@@ -48,6 +48,9 @@
         {{#link-to "ui-tooltip" class="list-group-item"}}
           Tooltip
         {{/link-to}}
+        {{#link-to "ui-slider" class="list-group-item"}}
+          Slider
```
Comment (zhujy8833) [inline tests/dummy/app/templates/application.hbs]: looks like we need to follow alphabetical order here?

File: addon/templates/components/ui-slider.hbs (hunk 1 above)
Comment (pzuraq) [inline addon/templates/components/ui-slider.hbs]: ++, you can insert bindings directly with HTMLbars now @dalvarez

File: addon/components/ui-slider.js (hunk 2 above)
Comment (pzuraq) [inline addon/components/ui-slider.js]: @twokul Looks like this is meant to be used only as a setter

File: tests/dummy/app/router.js [hunk 4]
```diff
@@ -20,4 +20,5 @@ export default Router.map(function() {
   this.route('ui-time-picker');
   this.route('ui-toggle');
   this.route('ui-tooltip');
+  this.route('ui-slider');
```
Comment (zhujy8833) [inline tests/dummy/app/router.js]: I guess we should also follow alphabetical order here :smiley_cat:

File: addon/components/ui-slider.js (hunk 2 above)
Comment (twokul) [inline addon/components/ui-slider.js]: ? I don't think so unless I'm missing something obvious `<input class="range-original" type="range" min={{min}} max={{max}} step={{step}} value={{value}} />`.

Comment (twokul) [inline addon/components/ui-slider.js]: rule of thumb is if you `get` something inside the computed, it should be a dependency key to avoid self trolling

Comment (pzuraq) [inline addon/components/ui-slider.js]: It's kind of hard to tell, the fact that value is passed in as a parameter makes me think that this is meant to be a guard if value is not set/doesn't exist yet. So return the minimum value OR the value on initialization. Confusing way to do it though, I would have used `on`, but this code was lifted directly from another ember addon we're using.

**Pull request merged**